presto-sqlserver/src/test/java/io/prestosql/plugin/sqlserver/TestSqlServerIntegrationSmokeTest.java

```diff
@@ -317,6 +324,27 @@ public void testNativeMultipleInClauses()
         sqlServer.execute("SELECT count(*) FROM dbo.orders WHERE " + longInClauses);
     }
 
+    @DataProvider
+    public Object[][] dataCompression()
+    {
+        return new Object[][] {
+                {NONE},
+                {ROW},
+                {PAGE}
+        };
+    }
+
+    @Test(dataProvider = "dataCompression")
+    public void testWithDataCompression(DataCompression dataCompression)
+    {
+        String tableName = "test_create_" + randomTableSuffix();
+        assertUpdate(format("CREATE TABLE %s (a BIGINT, b BIGINT) WITH (DATA_COMPRESSION = '%s')", tableName, dataCompression));
+        // Compression can be applied per partition so we might not be able to capture them in `SHOW CREATE TABLE`.
```

Comment:
How about verifying if all partitions have this enabled. If so you can set it to `true`, if all have disabled then `false`, otherwise you can return `null`. Do you think this is expensive to query `sys.partitions`?

Reply:
SQL Server connector is not aware of underlying partitioning. Why do we care in this case?

Reply:
I am thinking from Presto perspective, there is a table property that can be set via `CREATE TABLE` but `SHOW CREATE TABLE` is not returning it. Looks strange.

Reply:
i agree with this.

btw
what will happen if a new partition is added on sql server side?
will it implicitly copy compression attribute from some other partiiton?
otherwise, the "table property" can appear "unset"...

Reply:
I think it will refer to the compression type specified when creating a table

Reply:
> what will happen if a new partition is added on sql server side?

We should not care as long querying `sys.partitions` is a source of truth.

Reply:
> I think it will refer to the compression type specified when creating a table

where is it stored on sql server side?

Reply:
Can you add an explicit test for `SHOW CREATE TABLE` output?